server/src/test/java/org/elasticsearch/cluster/service/ClusterApplierServiceTests.java

```diff
@@ -66,6 +66,12 @@ public long relativeTimeInMillis() {
                 assertThat(Thread.currentThread().getName(), containsString(ClusterApplierService.CLUSTER_UPDATE_THREAD_NAME));
                 return currentTimeMillis;
             }
+
+            @Override
+            public long rawRelativeTimeInMillis() {
+                assertThat(Thread.currentThread().getName(), containsString(ClusterApplierService.CLUSTER_UPDATE_THREAD_NAME));
+                return currentTimeMillis;
+            }
```

Comment:
I notice that we use `rawRelativeTimeInMillis` in the `Recorder` but the threshold is in the order of seconds, I wonder if we could use `relativeTimeInMillis` instead (to reduce the calls to `System.nanoTime()`)

Reply:
The cached timer only updates every 200ms or so, and although the total threshold is many seconds in length many of the individual steps we record should take much less than 200ms. It's not a huge deal to call `System::nanoTime` here.